source/JarResolverLib/src/Google.JarResolver/PlayServicesSupport.cs

```diff
@@ -1038,7 +1036,10 @@ public void ClearDependencies()
                 KeyValuePair<Dependency, string> oldDepFilenamePair;
                 if (currentDepsByVersionlessKey.TryGetValue(dep.VersionlessKey,
                                                             out oldDepFilenamePair)) {
-                    if (oldDepFilenamePair.Key.BestVersion != dep.BestVersion &&
+                    string oldVersion = ExtractVersionFromFileName(
```

Comment:
Dependency.BestVersion should be the version string so in both cases this will be either a version string or an empty string see https://github.com/googlesamples/unity-jar-resolver/blob/master/source/JarResolverLib/src/Google.JarResolver/Dependency.cs#L158

Why did you need to change this to extract the version from the filename?  Are you seeing a file name sneak into this version field?